src/bin/shard/login.rs

```diff
@@ -449,3 +449,70 @@ pub fn buddy_menuchat_succ(clients: &mut ClientMap, state: &mut ShardServerState
 
     Ok(())
 }
+
+pub fn buddy_warp_succ(clients: &mut ClientMap, state: &mut ShardServerState) -> FFResult<()> {
+    let login_server = clients.get_login_server().unwrap();
+    let pkt: sP_LS2FE_REP_BUDDY_WARP_SUCC =
+        *login_server.get_packet(P_LS2FE_REP_BUDDY_WARP_SUCC)?;
+
+    let pc_id = pkt.iPC_ID;
+    let buddy_uid = pkt.iBuddyPCUID;
+    let buddy_shard = pkt.iShardNum;
+    let buddy_x = pkt.iBuddyX;
+    let buddy_y = pkt.iBuddyY;
+    let buddy_z = pkt.iBuddyZ;
+    let buddy_channel = pkt.iChannelNum;
+
+    let player = state.get_player_mut(pc_id)?;
+
+    player.set_position(Position {
+        x: buddy_x,
+        y: buddy_y,
+        z: buddy_z,
+    });
+
+    player.instance_id = rusty_fusion::chunk::InstanceID::default();
+
+    state.entity_map.update(
+        rusty_fusion::entity::EntityID::Player(pc_id),
```

Comment:
code style nit: import the struct and then do
```suggestion
        EntityID::Player(pc_id),
```